src/lib.rs

```diff
@@ -1,6 +1,560 @@
+}
+
+pub struct HammingDistance {}
+impl HammingDistance{}
+impl StringDiffAlgorithm for HammingDistance {
+    fn diff<'a>(&self, s1: &'a str, s2: &'a str) -> Vec<StringDiffOp>{
+        let mut opp_vec : Vec<StringDiffOp> = Vec::new();
+        if s1.len() != s2.len(){
+            panic!("Strings must be same length");
+        }
+        else{
+            for i in 0..s1.len(){
+                if s1.chars().nth(i).unwrap() != s2.chars().nth(i).unwrap(){
+                    let new_opp = StringDiffOp{
+                        kind : StringDiffOpKind::Substitute(
+                            s1.chars().nth(i).unwrap(),
+                            s2.chars().nth(i).unwrap() ),
+                        index : i as isize
+                    };
+                    opp_vec.push(new_opp)
+                }
+            }
+        }
+        return opp_vec;
+    }        
+
+    
+    fn distance<'a>(&self, s1: &'a str, s2: &'a str) -> usize{
+        let mut edit_distance = 0;
+        if s1.len() != s2.len(){
+            panic!("Strings must be same length");
+        }
+        else{
+            for i in 0..s1.len(){
+                if s1.chars().nth(i).unwrap() != s2.chars().nth(i).unwrap(){
+                    edit_distance += 1;
+                }
+            }
+        }
+        return edit_distance;
+    }
+}
+
+
+pub struct LevenshteinDistance {}
+impl LevenshteinDistance{
+    pub(crate) fn min_dist(x: usize, y: usize, z: usize) -> usize {
+        if x <= y && x <= z{
+            return x;
+        }
+        else if y <= x && y <= z {
+            return y;
+        }
+        else{
+            return z;
+        }
+    }
+
+    pub(crate) fn min_dist_with_dir(x: usize, y: usize, z: usize) -> (usize,char){
+        if x <= y && x <= z{
+            return (x, '^');
+        }
+        if y <= x && y <= z {
+            return (y, '<');
+        }
+        if z <= x && z <= y {
+            return (z, '\\');
+        }
+        return (0, 'x')
+    }
+
+    pub(crate) fn print_vector<T: std::fmt::Debug>(my_vector: &[T]){
+        for i in my_vector.iter(){
+            println!("{:?}", i);
+        }
+    }
+
+    pub(crate) fn get_operations(my_opp: &Vec<Vec<char>>, 
+                                    left_string: &str,
+                                    top_string: &str) -> Vec<StringDiffOp> {
+                        
+        let mut my_opp_vector : Vec<StringDiffOp> = Vec::new();
+        let mut top_str_len = top_string.len();
+        let mut left_str_len = left_string.len();
+        let mut prev_char : char = ' ';
+        
+        
+        loop{
+            
+            if top_str_len == 0 && left_str_len == 0{
+                break;
+            }
+                        //Rows               Columns
+            match my_opp[left_str_len][top_str_len] {
+                //insertion
+                '^' => {
+                    let insertionOpp : StringDiffOp = StringDiffOp{
```

Comment:
this should be `insertion_op` (variables are traditionally `camel_case`), also only 1 p, same for the instance below